platform/linux-generic/include/odp_buffer_internal.h

```diff
@@ -41,11 +41,19 @@ typedef struct seg_entry_t {
 	uint32_t  len;
 } seg_entry_t;
 
+typedef union buffer_index_t {
+	uint32_t u32;
+
+	struct {
+		uint32_t pool   :8;
+		uint32_t buffer :24;
```

Comment:
We originally had this index partitioning based on `ODP_CONFIG_POOLS`. Do we want to return to that here?

If not, we at least need an `ODP_STATIC_ASSERT()` to ensure that `ODP_CONFIG_POOLS < 256` or else bad things will happen here.

Reply:
OK, added checks in v2.